fondant/component.py

```diff
@@ -70,6 +80,14 @@ def _add_and_parse_args(self) -> argparse.Namespace:
 
         return parser.parse_args()
 
+    @property
+    def user_arguments(self) -> t.Dict[str, t.Any]:
```

Comment:
Maybe you can add a docstring here to specify that this is about the custom arguments that the user defined (excluding the Kubeflow arguments)